defender/downloaded/Contract_Sentinel_Governance.js

```diff
@@ -0,0 +1,220 @@
+        votes = `0.${'0'.repeat(18 - votes.length)}${votes[0]}`;
+      }
+      votes = internationalNumberFormat.format(votes);
+
+      proposalName = await getProposalTitle(proposalId);
+      if (displayName !== '') {
+        message = `**Vote** ${proposalName} ${supportEmoji} ${votes} by ${displayName} ${etherscanLink}`;
+      } else {
+        message = `**Vote** ${proposalName} ${supportEmoji} ${votes} by ${voter.slice(0, 6)} ${etherscanLink}`;
+      }
+
+      if (reason !== '') {
+        message += `\n\`\`\`${reason}\`\`\``;
+      }
+      break;
+    case 'ProposalCanceled':
+      ({ id } = params);
+      proposalName = await getProposalTitle(id);
+      message = `**Canceled Proposa**l ${proposalName} ${noEntryEmoji}`;
+      break;
+    case 'ProposalExecuted':
+      ({ id } = params);
+      proposalName = await getProposalTitle(id);
+      message = `**Executed Proposal** ${proposalName} ${checkMarkEmoji}`;
+      break;
+    case 'ProposalQueued':
+      ({ eta, id } = params);
+      proposalName = await getProposalTitle(id);
+      message = `**Queued Proposal** ${proposalName} ${checkMarkEmoji} available to execute at timestamp ${eta}`;
+      break;
+    default:
+      return undefined;
+  }
+
+  return message;
+}
+
+// eslint-disable-next-line func-names
+exports.handler = async function (autotaskEvent) {
+  console.log(autotaskEvent);
+  // ensure that the autotaskEvent Object exists
+  if (autotaskEvent === undefined) {
+    return {};
+  }
+
+  const { secrets } = autotaskEvent;
+  if (secrets === undefined) {
+    return {};
+  }
+
+  // ensure that there is a DiscordUrl secret
+  const { DiscordUrl: discordUrl } = secrets;
+  if (discordUrl === undefined) {
+    return {};
+  }
+
+  // ensure that the request key exists within the autotaskEvent Object
+  const { request } = autotaskEvent;
+  if (request === undefined) {
+    return {};
+  }
+
+  // ensure that the body key exists within the request Object
+  const { body } = request;
+  if (body === undefined) {
+    return {};
+  }
+
+  // ensure that the alert key exists within the body Object
+  const {
+    matchReasons,
+    hash: transactionHash,
+  } = body;
+  if (matchReasons === undefined) {
+    return {};
+  }
+
+  // create messages for Discord
+  const promises = matchReasons.map(async (reason) => {
+    // determine the type of event it was
+    const { signature, params } = reason;
+    const eventName = signature.slice(0, signature.indexOf('('));
+    // craft the Discord message
+    console.log('Creating Discord message');
+    return createDiscordMessage(eventName, params, transactionHash);
+  });
+
+  // wait for the promises to settle
+  const messages = await Promise.all(promises);
+
+  for (let i = 0; i < messages.length; i++) {
+    console.log('Posting to Discord');
+    console.log(messages[i]);
+    // eslint-disable-next-line no-await-in-loop
+    await postToDiscord(discordUrl, messages[i]);
+  }
```

Comment:
nit: replace the `for` loop with our more typical convention of using `.map` and `Promise.all`